### PR TITLE
Update MdDumper to use the new System.Reflection.Metadata package

### PR DESCRIPTION
**Pull request by cshung**
The [`Microsoft.Bcl.Metadata`](https://www.nuget.org/packages/Microsoft.Bcl.Metadata/1.0.9-alpha) package used in the original sample is alpha, ancient and unlisted.
Here is an update that uses the [`System.Reflection.Metadata`](https://www.nuget.org/packages/System.Reflection.Metadata) package instead.

I also made sure that the same code worked for both desktop and core.

@terrajobst @richlander @tmat

### CODE REVIEW

**Comment by cshung** [comment]
Ping @terrajobst @richlander @tmat

**Comment by cshung** [comment]
Ping @terrajobst @richlander @tmat

**Comment by cshung** [comment]
Ping @terrajobst @richlander @tmat

File: System.Reflection.Metadata/Desktop/MdDumper/MdDumper/MdDumper.csproj [hunk 1]
```diff
@@ -33,12 +34,25 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="System" />
-    <Reference Include="System.Collections.Immutable">
-      <HintPath>..\packages\Microsoft.Bcl.Immutable.1.1.20-beta\lib\portable-net45+win8\System.Collections.Immutable.dll</HintPath>
+    <Reference Include="System.Buffers, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
```
Comment (tmat) [inline System.Reflection.Metadata/Desktop/MdDumper/MdDumper/MdDumper.csproj]: I'd be better to use `PackageReference` for these.

Comment (tmat) [reply System.Reflection.Metadata/Desktop/MdDumper/MdDumper/MdDumper.csproj]: Also would be good to convert the project to SDK-style project.

**Comment by cshung** [comment]
@tmat, thanks for the review, and it is fixed.

File: System.Reflection.Metadata/Desktop/MdDumper/MdDumper/MdDumper.csproj [hunk 2]
```diff
@@ -1,80 +1,16 @@
...
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (tmat) [inline System.Reflection.Metadata/Desktop/MdDumper/MdDumper/MdDumper.csproj]: I think you can remove this.

**Comment by cshung** [comment]
> Actually, we shouldn't need two projects (one for Core and one for Desktop). Instead you can have one project that multi-targets.

Thanks, I will merge the projects.

**Comment by cshung** [comment]
@tmat, I don't have the permissions to merge this change.
If the code is ready, can you please merge it?

**Comment by tmat** [comment]
@cshung Someone who owns dotnet-samples should review and merge this - that's not me.

**Comment by cshung** [comment]
Thanks for the clarification, @tmat
@terrajobst @richlander, do we know who owns the repo? This is ready to be merged.

File: System.Reflection.Metadata/MdDumper/MdDumper.csproj [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>netcoreapp2.2;net461</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="System.Reflection.Metadata" Version="1.8.0-preview3.19551.4" />
```
Comment (terrajobst) [inline System.Reflection.Metadata/MdDumper/MdDumper.csproj]: For samples you might want to use the latest stable version, i.e. `1.7.0`

**Pull request merged**